DDG4/edm4hep/Geant4Output2EDM4hep.cpp

```diff
@@ -354,14 +354,11 @@ void Geant4Output2EDM4hep::saveParticles(Geant4ParticleMap* particles)    {
       const G4ParticleDefinition* def = p.definition();
       auto mcp = m_particles.create();
       mcp.setPDG(p->pdgID);
+      // Because EDM4hep is switching between vector3f[loat] and vector3d[ouble]
+      using MT = decltype(std::declval<decltype(std::declval<edm4hep::MCParticle>().getMomentum())>().x);
```

Comment:
```suggestion
      using MT = decltype(std::declval<edm4hep::MCParticle>().getMomentum().x);
```
I suppose you tried the version with `std::declval` less and it didn't work?

Reply:
I tried some version with a decl* less, but those didn't work.
Your proposal works.

Reply:
Potentially, we should also throw a `std::remove_cv_t<decltype(...)>` in there to get rid of a potential `const` that might be there because `getMomentum` will return `a const edm4hep::Vector3x&`. (Mabye we also need a `std::remove_reference_t` as well.

Can you check what `MT` is deduced to be here?

```cpp
template<typename T>
struct TypeDebug;
```
and then
```cpp
TypeDebug mtType<MT>{};
```
should lead to a compilation error that tells us the type.

Reply:
It should be safe without any of the `remove_cv` and `remove_reference` stuff: https://godbolt.org/z/o3zPcodd4

Reply:
Thanks for checking! Let's merge this then.